WIP: Lab 1

DanilByaliy opened the pull request:
Deployed at https://nodejs-labs-hw2i9sjax-danilbyaliy.vercel.app/api/hello

DanilByaliy committed:
npm init

DanilByaliy committed:
Add .gitignore and change type to module

DanilByaliy committed:
Add eslint

DanilByaliy committed:
Add Sonar-Lint to ESLint

DanilByaliy committed:
Add prettier and fix integration of prettier with eslint

DanilByaliy committed:
Add github actions for validation of code

DanilByaliy committed:
Add a simple endpoint

DanilByaliy committed:
Add pre-commit hooks

DanilByaliy committed:
Add Editorconfig

nosovk commented on .editorconfig:
```diff
@@ -0,0 +1,9 @@
+indent_size = 2
+end_of_line = lf
+trim_trailing_whitespace = true
+insert_final_newline = true
```
insert_final_newline = true
но самого newline что-то нет
добавь в prettier editorconfig: true

DanilByaliy replied:
added

nosovk commented on package.json:
```diff
@@ -0,0 +1,36 @@
+    "eslint-config-prettier": "^8.5.0",
+    "eslint-config-standard": "^17.0.0",
+    "eslint-plugin-import": "^2.26.0",
+    "eslint-plugin-n": "^15.3.0",
```
где этот плагин подключен?

nosovk commented on package.json:
```diff
@@ -0,0 +1,36 @@
+    "eslint-config-standard": "^17.0.0",
+    "eslint-plugin-import": "^2.26.0",
+    "eslint-plugin-n": "^15.3.0",
+    "eslint-plugin-promise": "^6.1.1",
```
и этот плагин нигде не подключен

DanilByaliy committed:
Set editorconfig true in .prettierrc.cjs

DanilByaliy committed:
Remove unused plugins

nosovk commented on .editorconfig:
```diff
@@ -0,0 +1,9 @@
+indent_size = 2
+end_of_line = lf
+trim_trailing_whitespace = true
+insert_final_newline = true
```
insert_final_newline = true
но как бы самого newline нету

DanilByaliy replied:
corrected

nosovk commented on package.json:
```diff
@@ -0,0 +1,34 @@
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "lint": "eslint --ignore-path .gitignore .",
+    "lint:fix": "npm run lint -- --fix",
```
то есть для проверки мы используем ignore-path, а для исправления нет?

DanilByaliy replied:
corrected

DanilByaliy committed:
Change some configurations

Add an empty final line to .editorconfig and an ignore-path pattern to run lint:fix

nosovk commented:
merge
